### PR TITLE
[STORM-2206] - initial checkin of visualization using vis.js

### PR DESCRIPTION
**Pull request by Crim**
Needs cleanup, UI work -- but its a functional POC using vis.js

### CODE REVIEW

**Comment by revans2** [comment]
My biggest concern right now it that it looks like you missed checking in a few files.

I see `vis.min.js`, `vis.min.css`, `streams.png`, and `bolt_info.png` all getting a 404 when I try to use it.

**Comment by Crim** [comment]
@revans2 You're right! (facepalm)  Updated the PR.

**Comment by revans2** [comment]
@Crim Don't feel bad.  When I was playing around with this I found (https://issues.apache.org/jira/browse/STORM-2526) which was my own face palm moment.

File: storm-core/src/ui/public/js/vis.min.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * * The Apache 2.0 License
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * and
+ *
+ * * The MIT License
+ *   http://opensource.org/licenses/MIT
+ *
+ * Vis.js may be distributed under either license.
```
Comment (revans2) [inline storm-core/src/ui/public/js/vis.min.js]: With this we need to make sure that we update LICENSE.txt to include this file and the license that we pick for it.

We probably also want to white list this and vis.min.css for rat, so it stops complaining about them.

**Comment by Crim** [comment]
Oh yea I should have mentioned that.  I had to go thru and fix that to get my local build working off of master :p

File: storm-core/src/ui/public/js/visualization.js [hunk 2]
```diff
@@ -16,434 +16,330 @@
...
-        $.each(jdat, function(k, v) {
-            for(var i = 0; i < v[":inputs"].length; i++)
-                if(v[":inputs"][i][":sani-stream"] === id) {
-                    
-                    sys.addEdge(v[":inputs"][i][":component"], k, v);
+
+        function updateAvailableStreams() {
+            var container = jQuery("#available-streams");
+            for (var streamName in availableStreamsHash) {
+                var entry = jQuery(container).find("#stream-" + streamName)
```
Comment (revans2) [inline storm-core/src/ui/public/js/visualization.js]: please test this out with a trident topology like org.apache.storm.starter.trident.TridentMinMaxOfDevicesTopology from the storm starter package.  The naming convention on the bolts/spouts is rather odd and messes with a lot of javascript/HTML.  Like for me I am seeing

```
Error: Syntax error, unrecognized expression: #stream-$batch  jquery-1.11.1.min.js:2:12720
	s</fb.error http://localhost:8080/js/jquery-1.11.1.min.js:2:12720
	s</fb.tokenize http://localhost:8080/js/jquery-1.11.1.min.js:2:18596
	fb http://localhost:8080/js/jquery-1.11.1.min.js:2:7068
	find http://localhost:8080/js/jquery-1.11.1.min.js:2:23593
	updateAvailableStreams http://localhost:8080/js/visualization.js:282:29
	addAvailableStream http://localhost:8080/js/visualization.js:275:17
	parseEdge http://localhost:8080/js/visualization.js:247:13
	parseNode http://localhost:8080/js/visualization.js:224:17
	parseResponse http://localhost:8080/js/visualization.js:101:17
	success http://localhost:8080/js/visualization.js:311:25
	j http://localhost:8080/js/jquery-1.11.1.min.js:2:27239
	fireWith http://localhost:8080/js/jquery-1.11.1.min.js:2:28057
	x http://localhost:8080/js/jquery-1.11.1.min.js:4:21841
	b http://localhost:8080/js/jquery-1.11.1.min.js:4:25897
```

I'm not sure what other places may be using stream/bolt names that need to be sanitized.

Comment (Crim) [inline storm-core/src/ui/public/js/visualization.js]: Ah I've never played with Trident, I'll dig into that and see what is going on and sanitize them where needed.  Thanks for the heads up. 👍 

I'll also dig into making the sizing of the slide outs dynamic.

**Comment by knusbaum** [comment]
Thanks a lot for this. It looks really good in general.
One thing I noticed that could use some improvement is the bolt-info dropdown.

The info can run off the window with no option to scroll. It seems like the scroll bar doesn't show up until the info is well out of view, and when it does show up, I can't scroll enough to bring everything back in view.

Two more minor things:
1. When clicking on a spout, the info window still says "Bolt Info"
2. It's not exactly clear what the things labeled "Up:" are.

**Comment by Crim** [comment]
Sorry guys, I barf'd that branch.  Closing this PR in favor of: https://github.com/apache/storm/pull/2136

I've made some changes, but its still a work in progress.  I need to take into account most of @knusbaum 's suggestions still.

**Pull request closed without merging**